dot/state/inmemory_storage.go

```diff
@@ -196,32 +199,8 @@ func (s *InmemoryStorageState) GetStorage(root *common.Hash, key []byte) ([]byte
 	return inmemory_trie.GetFromDB(s.db, *root, key)
 }
 
-// GetStorageByBlockHash returns the value at the given key at the given block hash
-func (s *InmemoryStorageState) GetStorageByBlockHash(bhash *common.Hash, key []byte) ([]byte, error) {
-	var (
-		root common.Hash
-		err  error
-	)
-
-	if bhash != nil {
-		header, err := s.blockState.GetHeader(*bhash)
-		if err != nil {
-			return nil, err
-		}
-
-		root = header.StateRoot
-	} else {
-		root, err = s.StorageRoot()
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	return s.GetStorage(&root, key)
-}
-
-// GetStateRootFromBlock returns the state root hash of a given block hash
-func (s *InmemoryStorageState) GetStateRootFromBlock(bhash *common.Hash) (*common.Hash, error) {
+// getStateRootFromBlock returns the state root hash of a given block hash
+func (s *InmemoryStorageState) getStateRootFromBlock(bhash *common.Hash) (*common.Hash, error) {
 	if bhash == nil {
 		b := s.blockState.BestBlockHash()
 		bhash = &b
```

Comment:
what if header comes back as nil on line 209?

Reply:
Do you mean `BlockState.GetHeader()` might return `header` and `err` as `nil`? 

At least in `DefaultBlockState` [that doesn't seem to be the case](https://github.com/ChainSafe/gossamer/blob/2196b5d94e0fa0fa8e8578e6d5f09bdec0a9f3cc/dot/state/block.go#L337).
If it can happen, this case wasn't handled before either. Should I add a check and return an error?